### PR TITLE
Added Scarlet notebook and outputs

### LINKED ISSUE #3: Fine-tune a Deepfake Video Detection Model [Scarlet]
We need to research, collect data, and fine-tune a deepfake detection model that works reliably on **videos**. This model will serve as the core engine for the Lopt backend.

## **Goals:**
- Select the best pre-trained base model for fine-tuning
- Choose a training platform (local or cloud-based)
- Prepare and preprocess the dataset
- Fine-tune the model with video input support
- Export the model in a format usable via FastAPI (e.g., ONNX, PyTorch `.pt`)

---

### 1.  Research and Select a Base Model
- [ ] Explore models on Hugging Face or research papers:
  - [DFDC](https://paperswithcode.com/sota/deepfake-detection-on-deepfake-detection)
  - `xception`, `EfficientNet`, `MesoNet`, `ConvNext`, `Vision Transformers (ViT)`
  - Hugging Face models: [xception-for-deepfake](https://huggingface.co/models?search=deepfake)
- [ ] Evaluate tradeoffs: Accuracy vs inference time
- [ ] Choose a model pretrained on image-level deepfake detection with support for transfer learning

### 2. Collect and Prepare Dataset
- [ ] Find and Download public datasets
- [ ] Write preprocessing script:
  - Extract frames from video
  - Detect & align faces
  - Apply augmentation (optional)

### 3. Fine-tune the Model
- [ ] Choose platform:
  - Google Colab, Kaggle, or local (if possible)
- [ ] Set up training pipeline:
  - Use `PyTorch` or `TensorFlow` Lightning
  - Data loader for frames or frame sequences
  - Loss function: Binary Cross-Entropy or Focal Loss
  - Evaluation metrics: Accuracy, AUC, F1
- [ ] Implement early stopping and checkpointing
- [ ] Train and log metrics

### 4. Evaluate and Export
- [ ] Evaluate model performance on validation and test sets
- [ ] Save best checkpoint
- [ ] Convert model to `.pt`, `.onnx`, or TensorFlow SavedModel format
- [ ] Benchmark inference speed

### 5. Organize Model and Metadata
- [ ] Save model in `.models/`
- [ ] Add `inference.py` script to load and run predictions
- [ ] Document input shape, preprocessing steps, and expected output format

---

## **Acceptance Criteria:**
- A fine-tuned model file is available in the repo.
- A reproducible training notebook.
- Evaluation metrics and dataset details are documented.
- Inference-ready code is integrated in `backend/services/deepfake_detector.py`

## **References:**
- [FaceForensics++](https://github.com/ondyari/FaceForensics)
- [Deepfake Benchmarks](https://paperswithcode.com/task/deepfake-detection)
- [Hugging Face Pretrained Models](https://huggingface.co/models)

### PR DESCRIPTION
**Pull request by y4sh-codes**
closes #3

### CODE REVIEW

**Review by Itz-Agasta — CHANGES_REQUESTED** [review]
OK looks good..... but  add a `requirements.txt` file in the video/ folder’s root. This will be important later when we containerize the app, and it’ll also help with notebook reproducibility so we don’t fall into `pip hell`

You can generate it from your Kaggle environment using:
```python
!pip freeze > requirements.txt
```
Then move that file into the video/ directory and commit it. Thanks!

**Comment by y4sh-codes** [comment]
Added the requirements.txt as asked

**Review by Itz-Agasta — COMMENTED** [review]
LGTM ...great work
